endpoints/openrtb2/auction.go

```diff
@@ -1664,8 +1664,27 @@ func validateRequestExt(req *openrtb_ext.RequestWrapper) []error {
 		}
 	}
 
-	if err := validateTargeting(prebid.Targeting); err != nil {
-		return []error{err}
+	if prebid.Targeting != nil {
+		if prebid.Targeting.PriceGranularity != nil {
+			if err := validatePriceGranularity(prebid.Targeting.PriceGranularity); err != nil {
+				return []error{err}
+			}
+		}
+		if prebid.Targeting.MediaTypePriceGranularity.Video != nil {
+			if err := validatePriceGranularity(prebid.Targeting.MediaTypePriceGranularity.Video); err != nil {
+				return []error{err}
+			}
+		}
+		if prebid.Targeting.MediaTypePriceGranularity.Banner != nil {
+			if err := validatePriceGranularity(prebid.Targeting.MediaTypePriceGranularity.Banner); err != nil {
+				return []error{err}
+			}
+		}
+		if prebid.Targeting.MediaTypePriceGranularity.Native != nil {
+			if err := validatePriceGranularity(prebid.Targeting.MediaTypePriceGranularity.Native); err != nil {
+				return []error{err}
+			}
+		}
 	}
```

Comment:
I recommend to keep the `validateTargeting` method and to extract this code block to that method. There's already a lot going in in `validateRequestExt` and this keeps it a bit simpler.